src/sidebar/config/fetch-config.js

```diff
@@ -212,11 +212,18 @@ export async function fetchConfig(appConfig, window_ = window) {
       requestConfigFromFrame.ancestorLevel,
       window_
     );
-    return await fetchConfigRpc(
+
+    const rpcMergedConfig = await fetchConfigRpc(
       appConfig,
       parentFrame,
       requestConfigFromFrame.origin
     );
+    // Add back the optional focused group id from the host page config
+    // as this may be needed for additional apps such aas the notebook.
+    return fetchConfigEmbed(
```

Comment:
Re-using `fetchConfigEmbed` here is risky IMO because the first argument is not the `appConfig`(ie. configuration from the page's HTML file) that `fetchConfigEmbed` is documented to accept. It is also not necessary because the result of `fetchConfigRpc` already includes the result of merging the app config (in the Notebook/Sidebar's HTML page) with the host config received via RPC.

I think what I would suggest doing instead is just explicitly merging the safe parts of `hostPageConfig` into `rpcMergedConfig`.